chronicles/textlineserializer.nim

```diff
@@ -26,47 +28,63 @@ serializationFormat TextLineLog,
                     mimeType = "text/plain"
 
 #
-# color support functions
+# value support functions
 #
-proc fgColor(writer: TextLineWriter, color: ForegroundColor, brightness: bool) =
-  when writer.colorScheme == AnsiColors:
-    writer.stream.writeText ansiForegroundColorCode(color, brightness)
-  when writer.colorScheme == NativeColors:
-    writer.stream.setForegroundColor(color, brightness)
-
-proc resetColors(writer: TextLineWriter) =
-  when writer.colorScheme == AnsiColors:
-    writer.stream.writeText ansiResetCode
-  when writer.colorScheme == NativeColors:
-    writer.stream.resetAttributes()
-
-proc applyStyle(writer: TextLineWriter, style: Style) =
-  when writer.colorScheme == AnsiColors:
-    writer.stream.writeText ansiStyleCode(style)
-  when writer.colorScheme == NativeColors:
-    writer.stream.setStyle({style})
+
+const
+  escChars: set[char] = strutils.NewLines + {'"', '\\'}
+  quoteChars: set[char] = {' ', '='}
+
+type
+  typesNotNeedingQuotes = float | float64 | float32 | uint8 | uint16 | uint32 | uint64 | int | int8 | int16 | int32 | int64
```

Comment:
This looks like `SomeNumber` (a similar union types defined in the system module)

Reply:
Good catch!